Feat(Notification): Add Notification APIs

MilanVaghasiya26 opened the pull request:
### Descriptions
This PR introduces two new APIs for managing user notifications:
1. Send Notification API – Allows sending notifications to users and storing them in the database.
2. Get All Notifications API – Retrieves a paginated list of notifications.

### Features Implemented
✅ POST **/notifications/send-notification** – Send a new notification.
- Accepts a JSON payload containing title, description, location, type, etc.
- Stores the notification in the database.
- Returns a success message upon successful creation.

✅ GET **/notifications/get-notifications?page=1&size=10** – Retrieve notifications with pagination.
- Implements pagination with page and size parameters.
- Returns a structured response including total count and has_more flag.

### Code Changes
- Added `NotificationController.ts ` for handling API requests.
- Added `NotificationService.ts` for business logic and database operations.
- Created `NotificationRequestDto.ts ` & `NotificationResponseDto.ts` for type safety.
- Updated `routes/notifications.ts` to define API endpoints.
- Implemented pagination logic using getPagination function for getAllNotifications.

MilanVaghasiya26 committed:
Feat(Notification):
- Create APIs for send and get all notifications
- Create request and response DTOs for notifications
- Change users table wishlist type from Reviews to Wishlist

abhaysinghs772 commented on src/routes/admin.ts:
```diff
@@ -1,13 +1,19 @@
 adminRouter.get('/customer/:customer_id/role/:user_role_type', adminController.getCustomerById);
 
+/* Notification-Mangement-section */
+adminRouter.post("/send-notification", [validateGetCustomersSchema], notificationController.sendNotification); // Send Notification
```
remove "validateGetCustomersSchema" and use proper validation

MilanVaghasiya26 replied:
Updated

abhaysinghs772 commented on src/routes/admin.ts:
```diff
@@ -1,13 +1,19 @@
 
+/* Notification-Mangement-section */
+adminRouter.post("/send-notification", [validateGetCustomersSchema], notificationController.sendNotification); // Send Notification
+adminRouter.get("/get-notifications", [validateGetCustomersSchema], notificationController.getAllNotifications);  // Get All Notifications (with pagination)
```
remove "validateGetCustomersSchema" and use proper validation

MilanVaghasiya26 replied:
Yes adding

abhaysinghs772 commented on src/entities/User.ts:
```diff
@@ -88,10 +95,7 @@ export class User extends BaseEntity {
   @OneToMany(() => Wishlist, (wishlist) => wishlist.user)
-  wishlist!: Reviews[];
-
-  // @OneToMany(() => LoginHistories, (loginHistory) => loginHistory.user)
```
please do not remove the comments, these are for future references

MilanVaghasiya26 replied:
Yes it was by mistake

abhaysinghs772 reviewed (changes requested):
please make your changes as per the review

MilanVaghasiya26 committed:
Feat(Notification):
- Add joi validation for get AND send notification API
- Add filters inside get all notification API

MilanVaghasiya26 committed:
Fix:
- Add removed comment code inside user entity

abhaysinghs772 reviewed (approved):
lgtm
